includes/polling/functions.inc.php

```diff
@@ -80,6 +80,13 @@ function poll_sensor($device, $class)
                 require 'includes/polling/sensors/' . $class . '/' . $device['os_group'] . '.inc.php';
             }
 
+            if (! is_numeric($sensor_value)) {
+                preg_match('/-?\d*\.?\d+/', $sensor_value, $temp_response);
```

Comment:
This regex is wrong. It would match `-.1` for example. 
```suggestion
                preg_match('/-?\d+(?:\.\d+)?/', $sensor_value, $temp_response);
```
Think this covers the cases

Reply:
Hi, I think -.1 is correct as a result of the regex and no change is needed. Also, this is a 1:1 copy of the same code in the discovery logic.
<?php
$value = -.1;
$type = gettype($value);
print $type;
$value++;
print "<br>";
print $value;
>>>
double
0.9